src/main/java/bio/terra/cli/command/auth/Status.java

```diff
@@ -1,38 +1,100 @@
 
 /** This class corresponds to the third-level "terra auth status" command. */
 @Command(name = "status", description = "Print details about the currently authorized account.")
-public class Status implements Callable<Integer> {
+public class Status extends CommandSetup {
 
-  @Override
-  public Integer call() {
-    GlobalContext globalContext = GlobalContext.readFromFile();
-    WorkspaceContext workspaceContext = WorkspaceContext.readFromFile();
+  @CommandLine.Mixin FormatFlag formatFlag;
 
+  /**
+   * Populate the current user in the global context and return an object with a subset of the
+   * TerraUser properties.
+   */
+  @Override
+  protected void execute() {
     new AuthenticationManager(globalContext, workspaceContext).populateCurrentTerraUser();
     Optional<TerraUser> currentTerraUserOpt = globalContext.getCurrentTerraUser();
 
     // check if current user is defined
+    AuthStatusReturnValue authStatusReturnValue;
     if (!currentTerraUserOpt.isPresent()) {
-      System.out.println("No current Terra user defined.");
-      return 0;
+      authStatusReturnValue = new AuthStatusReturnValue();
+    } else {
+      TerraUser currentTerraUser = currentTerraUserOpt.get();
+      authStatusReturnValue =
+          new AuthStatusReturnValue(
+              currentTerraUser.terraUserEmail,
+              currentTerraUser.terraProxyGroupEmail,
+              !currentTerraUser.requiresReauthentication());
+    }
+
+    formatFlag.printReturnValue(authStatusReturnValue, returnValue -> this.printText(returnValue));
+  }
+
+  /** POJO class for printing out this command's output. */
+  public static class AuthStatusReturnValue {
+    // Terra user email associated with the current user
+    public final String userEmail;
+
+    // Terra proxy group email associated with the current user
+    public final String proxyGroupEmail;
+
+    // true if the current user does not need to re-authenticate
+    public final boolean loggedIn;
+
+    // true if there is a current user defined in the global context
+    public final boolean currentUserDefined;
+
+    /** Constructor for when there is a current user defined. */
+    public AuthStatusReturnValue(String userEmail, String proxyGroupEmail, boolean loggedIn) {
+      this.userEmail = userEmail;
+      this.proxyGroupEmail = proxyGroupEmail;
+      this.loggedIn = loggedIn;
+      this.currentUserDefined = true;
     }
-    TerraUser currentTerraUser = currentTerraUserOpt.get();
-    System.out.println("Current Terra user: " + currentTerraUser.terraUserEmail);
-    System.out.println(
-        "Current Terra user's proxy group: " + currentTerraUser.terraProxyGroupEmail);
 
-    // check if the current user needs to re-authenticate (i.e. is logged out)
-    System.out.println("LOGGED " + (currentTerraUser.requiresReauthentication() ? "OUT" : "IN"));
+    /** Constructor for when there is NOT a current user defined. */
+    public AuthStatusReturnValue() {
```

Comment:
consider using a factory function for special constructions to make it more self documenting at the call site. e.g. `public static AuthStatusReturnValue createForNoDefinedUser()` or some better name.

Reply:
Updated to use a private constructor and public static methods.